### PR TITLE
Feature/aave stop orders

### PR DESCRIPTION
**Pull request by kjaredic**

### CODE REVIEW

File: contracts/triggers/AaveQuotePriceTrigger.sol
Comment (yippee-ki-yay) [inline contracts/triggers/AaveQuotePriceTrigger.sol:14]: Would rename to AaveV3QuotePriceTrigger to avoid confusion as we have a V2 and V3 ratio trigger as well, same for Trailing

File: contracts/triggers/AaveQuotePriceTrigger.sol
Comment (yippee-ki-yay) [inline contracts/triggers/AaveQuotePriceTrigger.sol:57]: Use TokenPriceHelper.getAddrForChainlinkOracle() instead of hardcoding it here

File: contracts/triggers/AaveTrailingQuotePriceTrigger.sol
Comment (yippee-ki-yay) [inline contracts/triggers/AaveTrailingQuotePriceTrigger.sol:194]: Use TokenPriceHelper.getAddrForChainlinkOracle() instead of hardcoding it here

File: contracts/triggers/AaveQuotePriceTrigger.sol
Comment (yippee-ki-yay) [inline contracts/triggers/AaveQuotePriceTrigger.sol:71]: This should be `baseTokenAddr` and `quoteTokenAddr` maybe better naming to avoid this kind of confusion

File: contracts/triggers/AaveQuotePriceTrigger.sol
Comment (yippee-ki-yay) [inline contracts/triggers/AaveQuotePriceTrigger.sol:64]: Do we even need this kind of token change if we are calling aave oracle and not chainlink directly?

File: contracts/triggers/AaveTrailingQuotePriceTrigger.sol
Comment (yippee-ki-yay) [inline contracts/triggers/AaveTrailingQuotePriceTrigger.sol:24]: natspec incorrect `quoteStartRoundId roundId of the quote token feed at time of subscription`

File: contracts/triggers/AaveTrailingQuotePriceTrigger.sol
Comment (yippee-ki-yay) [inline contracts/triggers/AaveTrailingQuotePriceTrigger.sol:194]: Use TokenPriceHelper.getAddrForChainlinkOracle() instead of hardcoding it here

File: contracts/triggers/AaveTrailingQuotePriceTrigger.sol
Comment (yippee-ki-yay) [inline contracts/triggers/AaveTrailingQuotePriceTrigger.sol:187]: Can use TokenPriceHelper.getRoundInfo() if we change the function to inject ` IAggregatorV3(aggregatorAddr)` as first param.

File: contracts/triggers/AaveTrailingQuotePriceTrigger.sol
Comment (yippee-ki-yay) [inline contracts/triggers/AaveTrailingQuotePriceTrigger.sol:95]: Why not put into a function so to make the flow more readable?

File: test/l2-strategies.js
Comment (yippee-ki-yay) [inline test/l2-strategies.js:311]: If we hardcode this no need to hardcode &nullAddress later, thats how we did it in other aaveV3 strategies

**Review by yippee-ki-yay — COMMENTED** [review]
Tests for trailing also needed before approval. Have a few bug fixes and recommendations that need to be handled as well.

File: contracts/triggers/AaveQuotePriceTrigger.sol
Comment (kjaredic) [reply contracts/triggers/AaveQuotePriceTrigger.sol]: you're right, we don't

File: contracts/triggers/AaveTrailingQuotePriceTrigger.sol
Comment (kjaredic) [reply contracts/triggers/AaveTrailingQuotePriceTrigger.sol]: good idea, it is done

File: contracts/triggers/AaveV3QuotePriceTrigger.sol
Comment (yippee-ki-yay) [inline contracts/triggers/AaveV3QuotePriceTrigger.sol:66]: Don't put WSTETH_ADDR and STETH_ADDR in TokenUtils. wrong place for it

File: contracts/triggers/AaveV3QuotePriceTrigger.sol
Comment (yippee-ki-yay) [inline contracts/triggers/AaveV3QuotePriceTrigger.sol:69]: We already have this sort of function at TokenPriceHelper.getWStEthPrice() but I'm not sure if it's even needed, you're not setting the token to steth at the beginning and there is not price for it in L2s

File: test/l2-strategies.js
Comment (yippee-ki-yay) [inline test/l2-strategies.js:366]: It's probably better to use SendTokenAndUnwrap()

**Review by yippee-ki-yay — CHANGES_REQUESTED** [review]
Has a few issues to be addressed before merge

File: contracts/triggers/AaveV3QuotePriceTrigger.sol
Comment (kjaredic) [reply contracts/triggers/AaveV3QuotePriceTrigger.sol:69]: this is true, will remove, changes needed if we plan to use this trigger on L1 in the future

File: test/l2-strategies.js
Comment (kjaredic) [reply test/l2-strategies.js]: agreed

**Review by yippee-ki-yay — APPROVED** [review]

**Pull request merged**
